optimizely/client/client.go

```diff
@@ -232,6 +255,41 @@ func (o *OptimizelyClient) GetAllFeatureVariables(featureKey string, userContext
 	return enabled, variableMap, err
 }
 
+func (o *OptimizelyClient) getExperimentDecision(experimentKey string, userContext entities.UserContext) (decisionContext decision.ExperimentDecisionContext, experimentDecision decision.ExperimentDecision, err error) {
+	defer func() {
+		if r := recover(); r != nil {
+			errorMessage := fmt.Sprintf(`optimizely SDK is panicking with the error "%s"`, string(debug.Stack()))
+			err = errors.New(errorMessage)
+			logger.Error(errorMessage, err)
+		}
+	}()
+
+	projectConfig, err := o.GetProjectConfig()
+	if err != nil {
+		logger.Error("Error retrieving project config", err)
+		return decisionContext, experimentDecision, err
+	}
+
+	experiment, err := projectConfig.GetExperimentByKey(experimentKey)
+	if err != nil {
+		logger.Error("Error retrieving experiment", err)
+		return decisionContext, experimentDecision, err
+	}
+
+	decisionContext = decision.ExperimentDecisionContext{
+		Experiment:    &experiment,
+		ProjectConfig: projectConfig,
+	}
+
+	experimentDecision, err = o.decisionService.GetExperimentDecision(decisionContext, userContext)
+	if experimentDecision.Variation != nil {
```

Comment:
kind of related to my previous comment, what is the relationship between err and experimentDecision.Variation? 
can you check for err instead of experimentDecision.Variation ?